src/__global.zig

```diff
@@ -47,9 +47,24 @@ pub fn setThreadName(name: StringTypes.stringZ) void {
     }
 }
 
-pub fn exit(code: u8) noreturn {
+/// Flushes output and exits with a success status code.
+pub fn exit() noreturn {
     Output.flush();
-    std.os.exit(code);
+    std.os.exit(0);
+}
+
+/// Flushes output and exits with a failure status code.
+pub fn crash() noreturn {
+    @setCold(true);
+    Output.flush();
+    std.os.exit(1);
+}
+
+/// Flushes output and exits otherwise unsuccessfully with a specific exit code.
+pub fn exitOther(exit_code: u8) noreturn {
```

Comment:
what do you think about renaming this to `existCustom` or `exitCode` to have more exact naming?

Reply:
I thought about using the word "custom" but "other" is shorter and not a less "exact" name IMO. It's the exit function you use for any *other* exit code or in any *other* case.

Reply:
It could also be named `crashOther` though.
I do think this API could still be improved. I still think an API consisting of `exit(.failure)`, `exit(.success)`, and `exit(.{ .other = 50 })` would be better but I wasn't sure if `@setCold(true)` would work properly (it seems mark the whole function as cold instead of only that code path? I think this could be fixed by putting the `@setCold(true)`s in other internal functions though) and I haven't received any feedback from Jarred.

Reply:
Ya I can agree on that, good point. But wouldn't making `exist_code` optional in `exit` more sens? so by default it would exist with 0 and if you pass any number in an argument it would use it 🤔

Reply:
@SerekKiri one problem with that is that Zig does not have optional parameters.

Reply:
Ok so I read wrong thread then, nvm! No more comments 👍🏼

Reply:
Oh I didn't see the comment about `crashOther` but that's the best option I think with `exist(.failure)z etc. but I guess @Jarred-Sumner feedback would help here. I'm not experienced with zig yet so I can't help with that too much. :godmode: